docs/adr/adr_improved_frontend_dx.md

```diff
@@ -0,0 +1,55 @@
+# ADR: Improved Frontend Developer Experience
+Decision Deadline: As soon as possible
+
+Author: [Felipe](https://github.com/fhenrich33)
+
+## Status
+
+Proposed, Implementing
+
+## Context
+Frontend development experience in its current form is quite lacking. Imposes a lot of artificial difficulties, tooling doesn't work as expected, and adds friction to our FE contributors.
+
+We want to make FE development better.
+
+## Decision drivers
+
+1. We must make FE onboarding and development seamless.
+2. Implementation must play well with the current FE setup (React/Vite).
```

Comment:
There are two implementation considerations we always consider with things like this, which is maintainability and longevity, since our team is rather small and can't afford to regularly upgrade. Ideally it would be 5 year longevity with minimum maintainability issues. Would be good if this could be addressed here and with all other tooling related ADRs - you'll see I made a similar comment on @pylipp prior ADRs.

Reply:
Are you looking for a more specific way of wording and addressing these concerns?

I ask because while I've written this ADR and the accompanying implementation (TBR soon) with maintainability and longevity in mind, I might not have eased concerns regarding those very valid points.

Reply:
Yes, sorry let me be more clear about the expectations here. Imperfect draft ADRs are certainly better than none at all, but the goal here is to fully capture the context, including the dimensions that were and were not considered. For environment changes in particular, I find it helpful to explicitly call out maintainability, scalability and power and how the options considered stack up. This is an old but still valid decision around engineering excellence to not necessarily go with the flashiest new thing, but the one that is the most maintainable yet powerful. 


An example of how this was recently handled for the FE bundler: https://github.com/boxwise/boxtribute/blob/master/docs%2Fadr%2Fadr_fe_bundler.md
This is a relatively "rough" writeup, but note that it explicitly covers maintainability and reliability. 

An more polished example from the BE where we explicitly cover reliability and power, and ultimately chose not to go with the gold standard: https://github.com/boxwise/boxtribute/blob/master/docs%2Fadr%2Fadr_Python-ORM.md I will note that despite its more polished state, it would be improved if maintainability waa covered in the same was as in the Vite writeup.

Reply:
> Implementation must play well with the current FE setup (React/Vite).

I'd like to add that it's probably preferred to avoid major friction when switching the tool stack (can't assess it myself from the proposed selection of tools), and it might be beneficial to do incremental updates if possible (again, can't assess that).

Reply:
@aerinsol I see. I'll make sure to expand on maintainability and scalability.

@pylipp Related to that, I picked these pieces as they are rather unobtrusive and play nice with what we currently have.

Reply:
@fhenrich33 great, if you've assessed it from a compatibility and upgradability standpoint, would be nice to see that explicitly called out. I would probably classify upgradability as a subset of maintainability.